Writer.py

```diff
@@ -19,40 +19,35 @@ def read_from_file(self):
             return students
 
     def main(self):
-        self.students = self.read_from_file()
-        self.students.sort(key=lambda k: k['Searches'], reverse=True)
-        student = {}
-        while len(self.students) < self.size:
-            student = {'Name': input('enter name : '), 'Searches': int()}
-            self.students.append(student)
-            self.write_to_file(self.students)
-
-        new = input('list full,add new, enter yes or no ?')
-        if new == 'yes':
-            self.update()
-
-    def update(self):
-        students = self.read_from_file()
-        students.sort(key=lambda k: k['Searches'])
-        print('update this',students)
-        if len(students) == 0:
+        while True:
+            input_name = input('enter name: ')
+            self.students = self.read_from_file()
+            if len(self.students) < self.size:
+                student = {'Name': input_name, 'Searches': int()}
+                self.students.append(student)
+                self.write_to_file(self.students)
+            else:
+                self.update(input_name)
+
+    def update(self, input_name):
+        self.students.sort(key=lambda k: k['Searches'])
+        print('update this', self.students)
+        if len(self.students) == 0:
             print('List is empty')
-            self.main()
         else:
-            add_new = input('please add new')
-            for student in students:
-                if student['Name'] != add_new:
-                    students.remove(student)
-                    add_item = {"Name": add_new, "Searches": 0}
-                    students.append(add_item)
-                    students.sort(key=lambda k: k['Searches'])
-                    self.write_to_file(students)
+            #add_new = input('please add new')
+            for student in self.students:
+                if student['Name'] != input_name:
+                    self.students.remove(student)
+                    add_item = {"Name": input_name, "Searches": 0}
+                    self.students.append(add_item)
+                    self.write_to_file(self.students)
                     break
 
-def call_writer():
-    size = int(input('what is size?'))
-    writer1 = Writer("instance writer1",size)
-    writer1.main()
+# while True:
```

Comment:
Please add if __name__ = "__main__" in both Reader and Writer... Other that that... all looks good.

Reply:
sorted